Update honeycomb instrumentation

poolski opened the pull request:
## Main Changes
This PR adds an optional field called `process` to the `setup()` function that apps call when instantiating `js-honeycomb-support` (e.g. [here](https://github.com/geckoboard/heirloom/blob/c9fc5fbd665387108e6721f55f3f03ebfd84badc/src/index.js#L38)). 

This field defaults to `http`, but is designed to allow the caller to tell Honeycomb what kind of process it is - e.g. a `http` server, `worker` process, etc.

## Other changes
I've bumped the version of the `honeycomb-beeline` package to conform with Honeycomb's minimum version requirement to work with Environments.

I've also removed the `yarn.lock` file as we [don't use it anymore](https://geckoboard.slack.com/archives/C03G07Q0G/p1724754307625039). I've replaced it with `package-lock.json` and updated CI to use `npm ci install` which [install packages specified in the lockfile](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable) instead of `package.json` for more reliable CI.

poolski committed:
Add `process` parameter to `setup()`

poolski committed:
Update tests

poolski committed:
Update beeline version to `3.3.0`

poolski committed:
Remove unused `yarn.lock`

poolski committed:
Add CI lint/test jobs

Cribbed from Concierge

poolski committed:
Update test & package cache steps in CircleCI

poolski committed:
Use Node20 in CI

poolski committed:
Update tests to conform with Honeycomb library

poolski committed:
Add `package-lock.json` to replace `yarn.lock`

poolski committed:
Use `npm ci` to install from lockfile

OliverKnights commented on index.js:
```diff
@@ -15,12 +15,13 @@ const instrumentHTTP = require('./instrumentation/http');
  * @returns {beeline.Beeline}
  */
-module.exports = function setup(name, gitSha, options) {
+module.exports = function setup(name, process = 'http', gitSha, options) {
```
Any reason this is the second arg? If it was optional and last wouldn't it mean we could update this package and nothing would break if the updater didn't notice the change (although maybe we only use this package in heirloom? so it doesn't really matter?)

poolski replied:
That's a good point! Is there a convention around adding new args to a method and where they should go?
I added it after `name` since it's related to the app, but there's no functional reason not to put it at the end of the list for the reason you suggested

OliverKnights replied:
I think it's typical to put optional parameters at the end of the argument list https://stackoverflow.com/questions/2896106/why-optional-parameters-must-appear-at-the-end-of-the-declaration

poolski committed:
Reorder `setup()` field names

This should make the `setup()` method compabible with callers which
don't explicitly specify their `process` type.
